### ISSUE TITLE
provider/postgis: The `id` field is not included as a property if specified in `fields` in the config

### ISSUE THREAD

**Issue by JohnnyRealImpact** [go-spatial/tegola#383]
# Context
When explicitly specifying the `id_fieldname` in the `fields` collection of a `config.toml` definition, then the `id` field is not included as a tag as expected. 

Example:
```
    [[providers.layers]]
    name = "geo_place"
    tablename = "metrics.dim_geo_places"
    id_fieldname = "id"
    geometry_fieldname = "shape"
    fields = [
        "id",    <--- This property is ignored...
        "name",
        "level",
        "valid_from",
        "valid_to"
    ]
```

I had a quick look at the source code and found the culprit here:
 - https://github.com/go-spatial/tegola/blob/6a08f2c826fe849dfd4571a58b4361c6c95aa7e0/provider/postgis/util.go#L169

I am no expert in go, but somehow I think that if the `id` field was explicitly specified in the `fields` array, then it should be included as a tag. My pseudo code solution would look something like this:
```
...
case idFieldname:
	gid, err = gId(v)
	if in_collection(descriptions, idFieldname) {
		tags[desc.Name] = value
	}
...
```

For now the work around is to explicitly define an additional column with another name, eg:
```
SELECT 
    gid, 
    ST_AsBinary(geom) as geom,
    id as my_id_column,     <---- this is the workaround...
FROM ...
WHERE geom && !BBOX!
```

Hope this makes sense, and helps.

**Comment by ARolek** [comment]
@JohnnyRealImpact apologies for the delay and thanks for the bug report and work around. I will try to get this fixed in our upcoming v0.7.0 release.

**Comment by JohnnyRealImpact** [comment]
Thank you, this is great news! :) 👍
